api/helpers/game-states/find-target-card.js

```diff
@@ -0,0 +1,57 @@
+const TargetType = require('../../../utils/TargetType.json');
+
+module.exports = {
+  friendlyName: 'Find target card',
+
+  description: "Finds card specified by its ID on the requested player's board. Returns null if not found",
+
+  inputs: {
+    targetId: {
+      type: 'string',
+      description: 'String ID of the card to find',
+      required: true,
+      example: 'AS', // Ace of Spades
+    },
+    targetType: {
+      type: 'ref', // TargetType enum
+      description: 'Whether the target card is a point card, face card, or jack',
+      default: null,
+      required: false,
+      example: TargetType.jack,
+    },
+    /**
+     * @param { Player } player - the player whose board should contain the target card
+     * @param { Card[] } player.hand - cards in the player's hand
+     * @param { Card[] } player.points - the player's point cards -- can contain attachments
+     * @param { Card[] } player.faceCards - the player's face cards
+     */
+    player: {
+      type: 'ref',
+      descriptions: 'Player object from the current game state',
+      required: true,
+    },
+  },
+  sync: true,
+  fn: ({ targetId, targetType, player }, exits) => {
+    switch (targetType) {
+      case 'point':
+        return exits.success(player.points.find(card => card.id === targetId));
```

Comment:
What happens if player is somehow undefined, will this still come through as a success?

Reply:
Yes atm it will succeed with a null value. My thought was to then catch that in the consuming validate helper (theoretically shouldn't happen in execute() because we'd already try in validate).

The idea is that callers do a null check like they do for the `.find()` calls on points etc:

```
          const targetCard = sails.helpers.gameStates.findTargetCard(
            requestedMove.targetId,
            requestedMove.targetType,
            opponent,
          );
          // Must have target
          if (!targetCard) {
            throw new Error(`Can't find the ${requestedMove.targetId} on opponent's board`);
          }
```

That said we could consider throwing in `findTargetCard()` and then we could even remove the `if (!targetCard)` from the consuming validate/execute helpers.

But my instinct is to keep it like this so that the error gets thrown in validate() itself. That way we can rely on `validate()` to be the place to look to see all the errors for any given move

Reply:
Okay that makes sense, I do think we should explicitly send back null if not found though because `.find` returns undefined if nothing is found.